python/lsst/pipe/analysis/visitAnalysis.py

```diff
@@ -436,7 +507,10 @@ def readCatalogs(self, dataRefList, dataset, repoInfo, aliasDictList=None):
         if not catList:
             raise TaskError("No catalogs read: %s" % ([dataRef.dataId for dataRef in dataRefList]))
 
-        return concatenateCatalogs(commonZpCatList), concatenateCatalogs(catList)
+        if self.config.hasFakes:
+            return concatenateCatalogs(commonZpCatList), concatenateCatalogs(catList), areaDict
+        else:
+            return concatenateCatalogs(commonZpCatList), concatenateCatalogs(catList)
```

Comment:
To avoid returning 2 or 3 things depending on what mode you run it in, would it make sense to return a Struct with areaDict = None if there's no areaDict? I also think it is worth making variable names for each of the things that is returned even if you don't want to Struct-ify it all. Either way, please update the return docstring appropriately.

Reply:
I'm also realizing that `calibrateCatalogs` should probably be updated to not use `fluxMag0` as it currently does, and instead to use the new `photoCalib` methods. Can you update that method for self-consistency with all the other great stuff you've done here?

Reply:
I think updating calibrate catalogs is also out of the scope of this ticket.
I have updated the docstring about areaDict.